hazelcast/src/test/java/com/hazelcast/map/nearcache/NearCacheTest.java

```diff
@@ -117,11 +117,18 @@ public void testBasicUsage() {
         }
 
         map.clear();
-        for (HazelcastInstance instance : instances) {
-            NearCache nearCache = getNearCache(mapName, instance);
-            int size = nearCache.size();
-            assertEquals("Near Cache size should be 0 but was " + size, 0, size);
-        }
+
+        assertTrueEventually(new AssertTask() {
```

Comment:
If we need an `assertTrueEventually` after a `map.clear()` now, this looks like a noticeable behavior change to me. Users might actually rely on the fact, that the Near Cache is already invalidated, when `map.clear()` returns. This will at least need a good documentation in the release notes.